src/construct.jl

```diff
@@ -483,7 +483,8 @@ function queuenext!(iter::ExprSplitter)
             newmod isa Module || throw(ErrorException("invalid redefinition of constant $(newname)"))
             mod = newmod
         else
-            if (id = Base.identify_package(mod, String(newname))) !== nothing && haskey(Base.loaded_modules, id)
+            id = Base.identify_package(mod, String(newname))
+            if id !== nothing && haskey(Base.loaded_modules, id)
```

Comment:
just a note for you: inference can't (back-)propagate type constraints  from `===` or `isa` in this kind of case:
```julia
if (x = y) isa T
    # some code assuming x::T
end
```